libraries/AP_FETtecOneWire/AP_FETtecOneWire.h

```diff
@@ -0,0 +1,253 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+    /* Protocol implementation was provided by FETtec */
+    /* Strongly modified by Amilcar Lucas, IAV GmbH */
+
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+
+#ifndef HAL_AP_FETTEC_ONEWIRE_ENABLED
+#define HAL_AP_FETTEC_ONEWIRE_ENABLED !HAL_MINIMIZE_FEATURES && !defined(HAL_BUILD_AP_PERIPH) && BOARD_FLASH_SIZE > 1024
+#endif
+
+#ifndef HAL_AP_FETTEC_ONEWIRE_GET_STATIC_INFO
+#define HAL_AP_FETTEC_ONEWIRE_GET_STATIC_INFO 0
+#endif
+
+#if HAL_AP_FETTEC_ONEWIRE_ENABLED
+
+#include <AP_ESC_Telem/AP_ESC_Telem.h>
+#include <AP_Param/AP_Param.h>
+
+
+
+class AP_FETtecOneWire : public AP_ESC_Telem_Backend
+{
+
+public:
+    AP_FETtecOneWire();
+
+    /// Do not allow copies
+    AP_FETtecOneWire(const AP_FETtecOneWire &other) = delete;
+    AP_FETtecOneWire &operator=(const AP_FETtecOneWire&) = delete;
+
+    static const struct AP_Param::GroupInfo var_info[];
+
+    /// periodically called from SRV_Channels::push()
+    void update();
+    static AP_FETtecOneWire *get_singleton() {
+        return _singleton;
+    }
+private:
+    static AP_FETtecOneWire *_singleton;
+    AP_HAL::UARTDriver *_uart;
+
+#if HAL_WITH_ESC_TELEM
+    static constexpr uint8_t MOTOR_COUNT_MAX = 4;//ESC_TELEM_MAX_ESCS; /// OneWire supports up-to 15 ESCs, but Ardupilot only supports 12

```

Comment:
i think this needs some comment/explanation of why 4 was chosen here, instead of   12 or 15 or ESC_TELEM_MAX_ESCS.

Reply:
That is fixed already